### PR TITLE
Tooltip component changes

### LINKED ISSUE #111: New Module: Tooltip
screen shot 2018-11-02 at 10 46 24 am

Related: https://github.com/eBay/ebayui-core/issues/67

https://ebay.invisionapp.com/share/CWN9X1JF6MA#/screens/299234476

### PR DESCRIPTION
**Pull request by prabhurangan**
## Description

DS6 tooltip component changes for below types:
- Tooltip (on-click & focusout)
- Infotip (on-click & actionable)
- Tourtip (displayed by default on the page & actionable)

## Context

This will be new component for DS6 (Tooltip)

## References

Fixes #111 

## Screenshots

Tooltip:
image

Infotip:
image

Tourtip:

Button Callout:
image

Link callout:
image

### CODE REVIEW

**Comment by msendlakowski** [comment]
The tooltip can be clipped by the edge of the viewport.  I can't think of a way to handle this in skin, but we need to take care of it when we build the component.

screen shot 2018-09-05 at 10 41 37 am

**Comment by msendlakowski** [comment]
@seangates The design system says we should support arrows in additional locations, like bottom right and bottom left.  I'm thinking we need this in skin, rather than just "tooltip__bottom-arrow".  What do you think?

**Comment by prabhurangan** [comment]
@msendlakowski Yes, it is supported. will update the PR with more screenshot

**Comment by seangates** [comment]
I think @prabhurangan has accounted for the following:

image

**Comment by PrashantAshok** [comment]
The `infotip` screenshot seems to be missing the close icon in your PR description.

File: src/less/tooltip/ds6/tooltip.less [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+    }
+
+    &__right-arrow,
+    &__right-toparrow,
+    &__right-bottomarrow {
+        margin-right: 16px;
+        right: 100%;
+
+        &::after {
+            box-shadow: 2px 2px 2px 0 #767676;
```
Comment (PrashantAshok) [inline src/less/tooltip/ds6/tooltip.less]: Replace hardcoded value with DS6 color variable.

File: src/less/tooltip/ds6/tooltip.less [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+
+    &__body {
+        margin: 16px 16px 16px 16px;
+    }
+}
+
+.tooltip {
+    &:hover {
+        .tooltip__top-arrow,
+        .tooltip__top-leftarrow,
```
[the remark this answers is not in the document]
Comment (seangates) [reply src/less/tooltip/ds6/tooltip.less]: @msendlakowski Mentioned by @PrashantAshok, above.

File: src/less/tooltip/ds6/tooltip.less [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+    &__left-toparrow,
+    &__left-bottomarrow,
+    &__right-arrow,
+    &__right-toparrow,
+    &__right-bottomarrow {
+        box-shadow: 0 0 5px 1px @ds6-color-g205-gray;
+        width: 344px;
+
+        &::after {
+            border-color: #fff;
```
Comment (PrashantAshok) [inline src/less/tooltip/ds6/tooltip.less]: Replace all hardocoded color values with variables.

File: src/less/tooltip/ds6/tooltip-base.less [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        width: 90%;
+    }
+}
+
+.div--link-callout {
+    margin: 0 auto;
+    width: 100%;
+}
+
+.margin-8 {
```
Comment (PrashantAshok) [inline src/less/tooltip/ds6/tooltip-base.less]: Are any of classes below this line being used?

Comment (prabhurangan) [reply src/less/tooltip/ds6/tooltip-base.less]: Yes, it is used

Comment (msendlakowski) [reply src/less/tooltip/ds6/tooltip-base.less]: It looks like these aren't actually used in the tooltip styling, just the docs.  If so, we should move these to the docs only.

Comment (msendlakowski) [reply src/less/tooltip/ds6/tooltip-base.less]: In fact, that applies to all classes in this file, like "width-100".  If it only applies to the docs then it shouldn't be part of the tooltip CSS.

File: src/less/tooltip/ds6/tooltip.less [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+        .tooltip__bottom-arrow,
+        .tooltip__bottom-leftarrow,
+        .tooltip__bottom-rightarrow,
+        .tooltip__left-arrow,
+        .tooltip__left-toparrow,
+        .tooltip__left-bottomarrow,
+        .tooltip__right-arrow,
+        .tooltip__right-toparrow,
+        .tooltip__right-bottomarrow {
+            visibility: visible;
```
Comment (msendlakowski) [inline src/less/tooltip/ds6/tooltip.less]: @seangates Can you provide input here.
This is a negative pattern according to a11y guidelines.
https://ebay.gitbooks.io/mindpatterns/content/disclosure/tooltip.html

We should be rendering the tooltip as visible content, and then hiding it using javascript.  That means its not a pure CSS component, but I'm thinking that's the right way to go though.  We'll have to give an example of the JS needed to hide/show the tooltip.

File: src/less/tooltip/ds6/tooltip.less (hunk 2 above)
[the remark this answers is not in the document]
Comment (msendlakowski) [reply src/less/tooltip/ds6/tooltip.less]: Shoot... I was 2 minutes too late.

File: src/less/tooltip/ds6/tooltip.less (hunk 5 above)
Comment (msendlakowski) [reply src/less/tooltip/ds6/tooltip.less]: This is definitely an issue and has to be rethought.  The current demo is not accessible and we're teaching them an anti-pattern.  The tooltip isn't keyboard accessible.

File: src/less/tooltip/ds6/tooltip.less [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+            left: 7%;
+        }
+    }
+
+    &__bottom-rightarrow,
+    &__top-rightarrow {
+        margin-left: -300px;
+
+        &::after {
+            left: 87%;
```
Comment (msendlakowski) [inline src/less/tooltip/ds6/tooltip.less]: According to the specs all these have to be fixed positions, not relative.  The ones on the right should have 8px of space between the arrow and the right edge, etc.

File: src/less/tooltip/ds6/tooltip.less [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+        &::after {
+            top: 65%;
+        }
+    }
+
+    &__window {
+        background-color: @ds6-color-p002-blue;
+        color: @ds6-color-g201-gray;
+        visibility: hidden;
+        width: 344px;
```
Comment (msendlakowski) [inline src/less/tooltip/ds6/tooltip.less]: We also need the 400px version for larger screens.  Use media-queries.  And can we use max-width instead of width?

File: src/less/tooltip/ds6/tooltip.less [hunk 8]
```diff
@@ -0,0 +1,404 @@
...
+        }
+    }
+
+    &__top-arrow,
+    &__top-leftarrow,
+    &__top-rightarrow {
+        margin-top: 16px;
+
+        &::after {
+            top: 2%;
```
Comment (msendlakowski) [inline src/less/tooltip/ds6/tooltip.less]: These can't be relative because the height of the box will vary based on content.  They should be fixed numbers.

File: tooltip.browser.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
+    "dependencies": [
+        {
+            "path": "./dist/tooltip/ds6/tooltip.css",
+            "if-flag": "skin-ds6"
```
Comment (seangates) [inline tooltip.browser.json]: We should consider how we deal with this in DS4. It's under Notice in DS4, and we might want to break out the DS4 tooltip from the DS4 notice so we can do the dependency for DS4 when folks are using the eBayUI component.

File: src/less/tooltip/ds6/tooltip.less (hunk 2 above)
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: `leftarrow` should be `left-arrow` as @PrashantAshok mentioned.

File: src/less/tooltip/ds6/tooltip.less (hunk 5 above)
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: Do not use `visibility: visible;` or `visibility: hidden;`. Please refer to:

Mind pattern:
- https://ebay.gitbooks.io/mindpatterns/content/disclosure/tooltip.html

Previous DS4 tooltip:
- https://github.com/eBay/skin/blob/master/src/less/notice/ds4/notice-base.less
- https://github.com/eBay/skin/blob/master/src/less/notice/ds4/notice-tooltip.less

File: src/less/tooltip/ds6/tooltip.less (hunk 8 above)
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: `top: 0;`

File: src/less/tooltip/ds6/tooltip.less [hunk 10]
```diff
@@ -0,0 +1,404 @@
...
+        }
+    }
+
+    &__left-arrow,
+    &__left-toparrow,
+    &__left-bottomarrow {
+        left: 110%;
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: Cannot be percentage based. Must be a `px` value. You could possibly use `calc(100% + 8px);` to create a generated `px` value.

File: src/less/tooltip/ds6/tooltip.less [hunk 11]
```diff
@@ -0,0 +1,404 @@
...
+    }
+
+    &__right-arrow,
+    &__right-toparrow,
+    &__right-bottomarrow {
+        margin-right: 16px;
+        right: 100%;
+
+        &::after {
+            left: 95%;
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: Cannot be percentage based. Must be a `px` value. You could possibly use `calc(100% - 8px);` to create a generated `px` value.

File: src/less/tooltip/ds6/tooltip.less [hunk 12]
```diff
@@ -0,0 +1,404 @@
...
+            margin-left: 5px;
+        }
+    }
+
+    &__bottom-arrow,
+    &__top-arrow {
+        margin-left: -161px;
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: Where did this number come from? It seems very odd. Try `calc(50% - 8px);` or something similar.

File: src/less/tooltip/ds6/tooltip.less [hunk 13]
```diff
@@ -0,0 +1,404 @@
...
+
+        &::after {
+            left: 45%;
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: Should probably be `calc(50% + 8px);`.

File: src/less/tooltip/ds6/tooltip.less [hunk 14]
```diff
@@ -0,0 +1,404 @@
...
+            margin-left: 5px;
+        }
+    }
+
+    &__bottom-leftarrow,
+    &__top-leftarrow {
+        margin-left: -24px;
+
+        &::after {
+            left: 7%;
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: `left: 8px;`

File: src/less/tooltip/ds6/tooltip.less (hunk 6 above)
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: Cannot be percentage based. Must be a `px` value. You could possibly use `calc(100% - 8px);` to create a generated `px` value.

File: src/less/tooltip/ds6/tooltip.less (hunk 7 above)
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: Please use `max-width`.

File: tooltip.browser.json (hunk 9 above)
Comment (seangates) [reply tooltip.browser.json]: See my requested changes comment.

File: src/less/tooltip/ds6/tooltip.less (hunk 7 above)
Comment (msendlakowski) [reply src/less/tooltip/ds6/tooltip.less]: I talked to @prabhurangan about this.  The problem with making it max-width rather than width is that it doesn't automatically flex to the max width since its absolutely positioned.  It just get's taller. There may be some workaround, but I haven't played with it.

screen shot 2018-09-06 at 3 06 13 pm

Comment (seangates) [reply src/less/tooltip/ds6/tooltip.less]: Oh. Maybe:

```
max-width: 344px;
width: 100%;
```

**Comment by seangates** [comment]
@msendlakowski @PrashantAshok After speaking with @prabhurangan we're going to take a different tack. Specifically, we're going to try and mimic as much of what the current DS4 CSS does, normalizing it, then customizing it to fit DS6.

We'll also keep it backward compatible regarding the DS4 class names, as we want to update to the new `.tooltip` class name. Right now, for example, if a static site wanted to move from DS4 to DS6, they would have to change _all their class names, too,_ which is pretty much a blocker.

So, Prabhu will make some progress today, and I will pair program with him tomorrow morning to round things out.

File: src/less/tooltip/ds6/tooltip.less (hunk 7 above)
Comment (msendlakowski) [reply src/less/tooltip/ds6/tooltip.less]: Tried that.  

But @prabhurangan  it looks like we may be able to change the structure a little.  If we add another container around it and that container is has a max-width of 344px.  Then the internal container doesn't have a max-width or width set.  It looks like that might work.

screen shot 2018-09-06 at 3 17 38 pm

screen shot 2018-09-06 at 3 18 02 pm

So:
```
<div style="position:absolute; max-width: 344px">
    <div style="position:absolute">
        <p>Content</p>
    </div>
</div>
```

**Comment by msendlakowski** [comment]
Sounds good Sean. Thanks.

**Comment by seangates** [comment]
@prabhurangan Seeing the left margin (and maybe the right margin?) is thinner than it used to be. The tooltip on the left is the **new** one, and the one on the right is the **current** one:

image

**Comment by seangates** [comment]
@prabhurangan The DS6 close icon doesn't appear to be in the correct place, per the spec:

image

image

**Comment by seangates** [comment]
@prabhurangan In fact, the content doesn't appear to be matching up vertically, either.

File: src/less/tooltip/ds6/tooltip.less [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+        font-size: @ds6-font-size-14;
+        padding: 16px 16px 16px 16px;
+        position: absolute;
+        z-index: 1;
+
+        &__close {
+            margin-top: 1px;
+        }
+
+        &__pointer-top-left,
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: I don't think the `__pointer` references are needed in DS6, correct?

File: src/less/tooltip/ds6/tooltip.less [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+
+    .infotip {
+        background-color: @ds6-color-g201-gray;
+        box-shadow: 0 0 5px 1px @ds6-color-g205-gray;
+        color: @ds6-color-g206-gray;
+
+        &__arrow-top-left,
+        &__arrow-top-center,
+        &__arrow-top-right {
+            box-shadow: -2px -2px 2px @ds6-color-g205-gray;
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: This box shadow seems a bit heavy. Let's ask design to get us the actual box shadow they used in a Zeplin project or something.

File: src/less/tooltip/ds6/tooltip.less [hunk 17]
```diff
@@ -0,0 +1,201 @@
...
+.tooltip-comp {
+    .tooltip,
+    .tourtip {
+        background-color: @ds6-color-p002-blue;
+        color: @ds6-color-g201-gray;
+    }
+
+    .infotip {
+        background-color: @ds6-color-g201-gray;
+        box-shadow: 0 0 5px 1px @ds6-color-g205-gray;
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: This box shadow seems a bit heavy. Let's ask design to get us the actual box shadow they used in a Zeplin project or something.

File: src/less/tooltip/ds6/tooltip.less [hunk 18]
```diff
@@ -0,0 +1,201 @@
...
+        &__arrow-right-bottom {
+            box-shadow: 2px -2px 2px @ds6-color-g205-gray;
+        }
+    }
+
+    .tooltip,
+    .infotip,
+    .tourtip {
+        font-size: @ds6-font-size-14;
+        padding: 16px 16px 16px 16px;
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: `padding: 16px;` is sufficient.

File: src/less/tooltip/ds6/tooltip.less [hunk 19]
```diff
@@ -0,0 +1,201 @@
...
+        position: absolute;
+        z-index: 1;
+
+        &__close {
+            margin-top: 1px;
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: Why is this needed? I don't see any reason to have this.

File: src/less/tooltip/ds6/tooltip.less [hunk 20]
```diff
@@ -0,0 +1,207 @@
...
+
+    .tooltip,
+    .infotip,
+    .tourtip {
+        font-size: @ds6-font-size-14;
+        padding: 16px;
+        position: absolute;
+        z-index: 1;
+
+        &__pointer-top-left,
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: Is there any reason to have `__pointer...` selectors.

File: src/less/tooltip/ds6/tooltip.less [hunk 21]
```diff
@@ -0,0 +1,207 @@
+@import "../../less/ds6/variables.less";
+@import "../../less/ds6/mixins.less";
+
+@import "tooltip-mixins.less";
+@import "../base/tooltip.less";
+
+.tooltip-comp {
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: Is `.tooltip-comp` only for the documentation? If so, it needs to be removed.

File: src/less/tooltip/ds6/tooltip.less [hunk 22]
```diff
@@ -0,0 +1,207 @@
...
+    }
+
+    &:focus,
+    &:hover {
+        color: @ds6-color-link-default;
+        cursor: pointer;
+    }
+}
+
+button.tourtip__callout-close {
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: This could probably just be `.tourtip__cta`.

File: src/less/tooltip/ds6/tooltip.less [hunk 23]
```diff
@@ -4,151 +4,153 @@
...
-        }
-
-        p {
-            margin: 0;
-
-            &:not(:first-of-type) {
-                margin-top: 4px;
-            }
+        &:not(:first-of-type) {
+            margin-top: 4px;
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: I actually think this could be `0.25em` since it's based on the line-height of the paragraphs. Thoughts?

File: dist/tooltip/ds6/tooltip.css [hunk 24]
```diff
@@ -198,9 +252,15 @@
...
-.tooltip-comp .tourtip__arrow-right-bottom {
+.tooltip--tooltip .tooltip__arrow-right-bottom,
+.tooltip--infotip .tooltip__arrow-right-bottom,
+.tooltip--tourtip .tooltip__arrow-right-bottom,
+.tooltip--tooltip .infotip__arrow-right-bottom,
+.tooltip--infotip .infotip__arrow-right-bottom,
+.tooltip--tourtip .infotip__arrow-right-bottom,
+.tooltip--tooltip .tourtip__arrow-right-bottom,
+.tooltip--infotip .tourtip__arrow-right-bottom,
+.tooltip--tourtip .tourtip__arrow-right-bottom {
```
Comment (seangates) [inline dist/tooltip/ds6/tooltip.css]: This feels like a little excess duplication. We can probably get away with simply:
```
.tooltip .arrow-right-bottom
```
Then we can simplify the HTML, too by always referring to the arrow positions no matter the variation we're talking about. It's a much better use of the CSS cascade, too.

Thoughts?

File: src/less/tooltip/ds6/tooltip.less [hunk 25]
```diff
@@ -0,0 +1,205 @@
...
+            &__arrow-top-center {
+                .tooltip-arrow-top;
+
+                left: calc(50% - 8px);
+            }
+
+            &__arrow-top-right {
+                .tooltip-arrow-top;
+
+                right: 12px;
```
Comment (msendlakowski) [inline src/less/tooltip/ds6/tooltip.less]: We should apply `12px` as a variable, then it can be applied and updated consistently.  (For arrow-bottom-left it is 16px right now instead of 12 like the others)

**Comment by seangates** [comment]
Wondering if there is an issue with a selector because the box shadow is missing for the tooltip and tourtip (but is working for the infotip):

image

File: dist/tooltip/ds6/tooltip.css (hunk 24 above)
Comment (msendlakowski) [reply dist/tooltip/ds6/tooltip.css]: I agree.  In the specs the arrow position doesn't change based on the type of tooltip.  So no need to duplicate the arrow styles for each type.

**Comment by seangates** [comment]
Please add a `/src/less/tooltip/ds6/tooltip-mixin.less` and pattern it after the one in `/src/less/tooltip/ds6/textbox-mixin.less`.

Also, you'll need to create a `tooltip.browser.json` so that the eBayUI core components can pick it up using `@ebay/skin/tooltip` through Lasso.

**Comment by seangates** [comment]
Also ... add a reference in `gh.browser.json`. You'll see that pattern established there.

File: gh.browser.json [hunk 26]
```diff
@@ -18,6 +18,7 @@
         "less-import: ./src/less/radio/ds6/radio-mixin.less",
         "less-import: ./src/less/select/ds6/select-mixin.less",
         "less-import: ./src/less/textbox/ds6/textbox-mixin.less",
+        "less-import: ./src/less/tooltip/ds6/tooltip-mixin.less",
```
Comment (seangates) [inline gh.browser.json]: I see the reference here, but the file doesn't exist in this PR.

File: docs/_includes/ds6/tooltip-custom.html [hunk 27]
```diff
@@ -0,0 +1,157 @@
+<p>There are 3 types of tooltips available: traditional tooltips, infotips, and tourtips</p>
+<p>Tooltips pointer can be positioned using any one of the following elements based on the type (tooltip/infotip/tourtip): 
+    <span class="highlight">tooltip__arrow-top-left</span>,
```
Comment (seangates) [inline docs/_includes/ds6/tooltip-custom.html]: Should we just make this a `<ul>` with `<li>`'s?

Comment (seangates) [reply docs/_includes/ds6/tooltip-custom.html]: With the spans for highlighting still, of course. Just giving a bulleted list might help our developers.

File: src/less/tooltip/ds6/tooltip.less [hunk 28]
```diff
@@ -199,7 +206,24 @@ button.tourtip__cta {
...
 
 @media (min-width: 601px) {
-    .tooltip__content {
-        max-width: 400px;
+    .tooltip {
+        // &--tooltip,
+        // &--infotip,
+        // &--tourtip {
+        //     max-width: 400px;
+        // }
```
Comment (seangates) [inline src/less/tooltip/ds6/tooltip.less]: Remove all comments which aren't used.

File: src/less/tooltip/ds4/tooltip-mixin.less [hunk 29]
```diff
@@ -0,0 +1,3 @@
+.tooltip-mixin() {
+    @import "../../../../dist/tooltip/ds4/tooltip.css";
+}
```
Comment (seangates) [inline src/less/tooltip/ds4/tooltip-mixin.less]: There's no DS4 need for this file. Only for DS6. Please delete it.

File: docs/_includes/ds6/tooltip-custom.html [hunk 30]
```diff
@@ -6,18 +6,18 @@
...
+    <span class="highlight">arrow-top-center</span>, 
+    <span class="highlight">arrow-bottom-top</span>, 
+    <span class="highlight">arrow-bottom-right</span>, 
+    <span class="highlight">arrow-bottom-center</span>, 
+    <span class="highlight">arrow-left-top</span>, 
+    <span class="highlight">arrow-left-bottom</span>, 
+    <span class="highlight">arrow-left-center</span>, 
+    <span class="highlight">arrow-right-top</span>, 
+    <span class="highlight">arrow-right-center</span>, 
+    <span class="highlight">arrow-right-bottom</span>
```
Comment (seangates) [inline docs/_includes/ds6/tooltip-custom.html]: UL here? With each line being:
```
<li><span class="highlight">arrow-right-bottom</span></li>
```
Thoughts?

File: docs/static/ds6/skin.css [hunk 31]
```diff
@@ -2881,42 +2881,66 @@ span.textbox__icon:first-child,
...
 .tooltip--infotip .arrow-left-top,
+.tooltip--tourtip .arrow-left-top,
+.tooltip--tooltip .arrow-left-center,
 .tooltip--infotip .arrow-left-center,
-.tooltip--infotip .arrow-left-bottom {
-  box-shadow: -2px 2px 2px #767676;
+.tooltip--tourtip .arrow-left-center,
+.tooltip--tooltip .arrow-left-bottom,
+.tooltip--infotip .arrow-left-bottom,
+.tooltip--tourtip .arrow-left-bottom {
```
Comment (seangates) [inline docs/static/ds6/skin.css]: Hmmm. Seems to have bloated. Can you have a look as to your selector nesting in the LESS?

This should really be simply:
```
.tooltip .arrow-left-top,
.tooltip .arrow-left-center,
.tooltip .arrow-left-bottom { ... }
````
Thoughts?

File: dist/notice/ds4/notice.css [hunk 32]
```diff
@@ -56,28 +56,17 @@ button.page-notice__close span {
   display: inline-block;
   vertical-align: middle;
 }
-.flyout-notice,
```
Comment (ianmcburnie) [inline dist/notice/ds4/notice.css]: @seangates @prabhurangan Removing these `.flyout-notice` classes is a breaking change to the Notice module. I don't understand why they would be moved to the Tooltip module either.

Comment (ianmcburnie) [reply dist/notice/ds4/notice.css]: Oh, never mind, I see all those rules were moved within a media query.

**Comment by ianmcburnie** [comment]
For DS6, I feel like we should have stuck with the existing DS4 name of `flyout-notice`, because the disparity now is quite confusing. In DS4, a tooltip was always a *type* of flyout-notice, but we've now lost that association in DS6.
